client/lib/protect-form/index.jsx

```diff
@@ -15,8 +15,37 @@ const confirmText = i18n.translate( 'You have unsaved changes. Are you sure you
 const beforeUnloadText = i18n.translate( 'You have unsaved changes.' );
 let formsChanged = [];
 
+function warnIfChanged( event ) {
+	if ( ! formsChanged.length ) {
+		return;
+	}
+	debug( 'unsaved form changes detected' );
+	( event || window.event ).returnValue = beforeUnloadText;
+	return beforeUnloadText;
+}
+
+function markChanged( form ) {
+	if ( ! includes( formsChanged, form ) ) {
+		formsChanged.push( form );
+	}
+}
+
+function markSaved( form ) {
+	formsChanged = without( formsChanged, form );
+}
+
+/*
+ * HOC that passes markChanged/markSaved props to the wrapped component instance
+ */
 export const protectForm = WrappedComponent => {
 	return class ProtectedFormComponent extends Component {
+		// This function binding makes the listener attached to window unique for each
+		// component instance.
+		warnIfChanged = warnIfChanged.bind();
```

Comment:
The only purpose of this `bind` call was to create an unique function object, so that `addEventListener`/`removeEventListener` work correctly. But I rewrote this code completely, there's no longer a need to call `bind` here.